docs/frontend-system/building-plugins/01-index.md

```diff
@@ -6,50 +6,153 @@ sidebar_label: Overview
+  "exports": {
+    ...
+    "./alpha": "./src/alpha.ts"
+    ...
+  }
+```
+
+Note that we export the plugin as the default export of our package from `src/index.ts`. This is important using the new
+feature discovery of the new frontend system. This default export is how users of our plugin are able to seamlessly install
+the plugin package in a Backstage app without having to reference the plugin instance through code.
+
+## Registering a Plugin with your Backstage App
+
+The new frontend plugin system is designed to use the optional [feature discovery](../architecture/10-app#feature-discovery) for registration. Please see the
+documentation on Feature Discovery for more information and how to turn it on.
+
+However, if you wish to manually register a plugin rather than using automated Feature Discovery, plugins are now designed
+to be passed into the `features` field of the `createApp({...})` options.
+
+Manual Plugin Registration:
+
 ```tsx title="in src/index.ts"
-export { examplePlugin as default } from './plugin';
+import { createApp } from '@backstage/frontend-defaults';
+import myExamplePlugin from '@just-an-example/alpha';
+
+// Create your app instance
+const app = createApp({
+  // Features such as plugins can be installed explicitly, but we will explore other options later on
+  features: [myExamplePlugin],
+});
 ```
 
-Note that we export the plugin as the default export of our package from `src/index.ts`. This is important, as it is how users of our plugin are able to seamlessly install the plugin package in a Backstage app without having to reference the plugin instance through code.
+This will register our new completely empty and useless plugin with Backstage. You will need to add features via extensions.
 
-The plugin ID should be a lowercase dash-separated string, while the plugin instance variable should be the camel case version of the ID with a `Plugin` suffix. For more details on naming patterns within the frontend system, see [the article on naming patterns](../architecture/50-naming-patterns.md). By sticking to these naming patterns you ensure that users of your plugin more easily recognize the exports and features provided by your plugin.
+## Adding Plugin Features (via Extensions)
 
-## Adding extensions
+To add functionality to a plugin you need to add at least one [extension](../architecture/20-extensions.md). Let's continue by adding a
+standalone page to our plugin, as well as a navigation item that allows users to navigate to the page.
 
-The plugin that we created above is empty, and doesn't provide any actual functionality. To add functionality to a plugin you need to create and provide it with one or more [extensions](../architecture/20-extensions.md). Let's continue by adding a standalone page to our plugin, as well as a navigation item that allows users to navigate to the page.
+Extensions can be created directly by following the `ExtensionDefinition` type or methods like `createComponentExtension()`,
```

Comment:
Don't think we want to promote `ExtensionDefinition` type at all here, don't craft them yourself.

Use either `Blueprints` or `createExtension` directly is probably the wording we want.